velox/functions/sparksql/window/NthValue.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/common/base/Exceptions.h"
+#include "velox/exec/WindowFunction.h"
+#include "velox/expression/FunctionSignature.h"
+#include "velox/functions/lib/window/NthValueBase.h"
+#include "velox/vector/FlatVector.h"
+
+namespace facebook::velox::functions::window::sparksql {
+
+namespace {
+class NthValueFunction : public functions::window::NthValueFunctionBase {
+ public:
+  NthValueFunction(
+      const std::vector<exec::WindowFunctionArg>& args,
+      const TypePtr& resultType,
+      velox::memory::MemoryPool* pool)
+      : NthValueFunctionBase(args, resultType, pool) {
+    VELOX_USER_CHECK(
+        args[1].type->isInteger(), "Offset must be Integer for spark");
+    VELOX_USER_CHECK(args[1].constantValue, "Offset must be literal for spark");
+    if (args[1].constantValue->isNullAt(0)) {
+      setIsConstantOffsetNull(true);
+      return;
+    }
+    std::optional<vector_size_t> constantOffset =
+        args[1]
+            .constantValue->template as<ConstantVector<vector_size_t>>()
+            ->valueAt(0);
+    setConstantOffset(constantOffset);
+    VELOX_USER_CHECK_GE(constantOffset.value(), 1, "Offset must be at least 1");
+  }
+};
+} // namespace
+
+void registerIntegerNthValue(const std::string& name) {
+  std::vector<exec::FunctionSignaturePtr> signatures{
```

Comment:
Add a description of the signature like
// nth_value(T, integer) -> T.

Reply:
Updated, thanks.